### ISSUE TITLE
Operator overloading error when building test drivers

### ISSUE THREAD

**Issue by ashermancinelli** [LLNL/hiop#110]
Similar to issue #109, several issues have appeared on develop branch. Some of these are resolved in branch `ci-cmake-fix#107`, however some remain, such as this:

```
/people/manc568/projects/hiop/tests/LinAlg/matrixTestsSparse.hpp(765): error: no operator "++" matches these operands
            operand types are: std::initializer_list<int> ++

/people/manc568/projects/hiop/tests/LinAlg/matrixTestsSparse.hpp(777): error: no operator "<" matches these operands
            operand types are: std::initializer_list<int> < const hiop::tests::local_ordinal_type

/people/manc568/projects/hiop/tests/LinAlg/matrixTestsSparse.hpp(778): error: no operator "+" matches these operands
            operand types are: std::initializer_list<int> + hiop::tests::local_ordinal_type

/people/manc568/projects/hiop/tests/LinAlg/matrixTestsSparse.hpp(795): error: no operator "+" matches these operands
            operand types are: hiop::tests::local_ordinal_type + std::initializer_list<int>

/people/manc568/projects/hiop/tests/LinAlg/matrixTestsSparse.hpp(796): error: no operator "+" matches these operands
            operand types are: hiop::tests::local_ordinal_type + std::initializer_list<int>
```

This occurs when building with `HIOP_SPARSE=OFF`, `HIOP_USE_RAJA=OFF`, and `HIOP_USE_UMPIRE=OFF`.

Branched off of issue #107.

CC @pelesh @cnpetra @nychiang

**Comment by pelesh** [comment]
I can reproduce this with `HIOP_SPARSE=ON`, `HIOP_USE_GPU=ON` `HIOP_USE_RAJA=ON`, and `HIOP_USE_UMPIRE=ON`.

**Comment by ashermancinelli** [comment]
To add on, I reproduce using all options in the build script

**Comment by pelesh** [comment]
Local counter on line 761 in `matrxTestsSparse.hpp` is declared as:
```c++
auto nnz_A_need_to_copy{0};
```
Apparently not every compiler can infer that the type of `nnz_A_need_to_copy` should be `int`. Rewriting this as
```c++
local_ordinal_type nnz_A_need_to_copy = 0;
```
resolves this problem.

**Comment by ashermancinelli** [comment]
Perhaps we should add something to the style guides along these lines? I think `auto` should only be used when the type is immediately obvious, eg
```cpp
auto* v2 = dynamic_cast<T*>(v1);
```
`v2` is clearly of type T.

**Comment by cnpetra** [comment]
fixed by pr #120
